packages/cozy-intent/readme.md

```diff
@@ -0,0 +1,122 @@
+# Cozy-intent
+
+## Concept
+
+This library allows a react-native and a react app running in a webview to communicate in unidirectional and/or bidirectional fashions. The library is following declarative paradigms and is fully strictly typed. An emphasis has been put on low configuration needs and ease of use.
+
+In a nutshell, one end says what it wants to an other end, and the other end handles it any way it wants. For more information on the underlying systems of this library, see https://github.com/alesgenova/post-me.
+
+## Installation
+
+The library needs to be installed both in the parent and the children application.
+
+- `yarn add cozy-webview-intent`
+
+## API
+
+### Nomenclature
+
+- `event`: simple function that sends an event to the listening messenger.
+- `intent`: high-level abstraction intended to describe the whole flow of events/methods and the actual result of those calls in the application.
+- `messenger`: class implementing `post-me` messenger paradigm. Messengers are environment bound (DOM, React-native, worker, iframe, etc). They send and listen methods and events to other connected messengers.
+- `method`: messenger function that accept simple arguments to be handled by the listening messenger.
```

Comment:
I'm wondering if we can find a "less-generic" name for those `methods` to prevent ambiguity with "language" `methods`? Maybe something like `target methods` or `registered methods` (this one is used on your doc)

Reply:
yeah, I used post-me naming convention, which uses events and methods. But the more I think about it, the less it makes sense. You can have methods with no arguments, and that's functionally the same as sending an event. Maybe we should abstract completely those post-me internals and use a single way of communication. I see your suggestions but I think it should be a single word.